### PR TITLE
Embroider optimized compatibility

### PR DESCRIPTION
**Pull request by nickschot**
- [x] Write a README section on how to use it in an embroider optimized app (i.e. pass imported component instead of string)

### CODE REVIEW

**Comment by nickschot** [comment]
This needs #385 before we can merge it so that support is actually tested

File: addon/templates/components/modal.hbs [hunk 1]
```diff
@@ -1,4 +1,6 @@
 {{!-- template-lint-disable no-action --}}
 <div id={{this.modalElementId}} class="epm-modal {{this.optionsClassName}} {{this.animatingOutClass}}">
-  {{component @modal._name data=@modal._data close=(action "close")}}
+  {{#let (component (ensure-safe-component @modal._name) data=@modal._data close=(action "close")) as |ModalComponent|}}
```
Comment (pichfl) [inline addon/templates/components/modal.hbs:3]: Should we consider actually deprecating passing in components by name?

Comment (nickschot) [reply addon/templates/components/modal.hbs:3]: We can't yet as there is no functional alternative for non-embroider yet as far as I'm aware (it was planned though iirc).

**Review by pichfl — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -40,6 +41,8 @@
     "@babel/eslint-parser": "^7.16.0",
     "@ember/optional-features": "^2.0.0",
     "@ember/test-helpers": "^2.6.0",
+    "babel-eslint": "^10.1.0",
```
Comment (Turbo87) [inline package.json:44]: this seems unrelated, no? also not alphabetically ordered 😉

Comment (Turbo87) [reply package.json:44]: yeah, no, this shouldn't be here since we're using `@babel/eslint-parser` above 🤔

Comment (nickschot) [reply package.json:44]: Ah good catch! Not sure anymore which one we need. I'll check it out later and fix it.
